### ISSUE TITLE
Recording daemon stops on lipstick restart

### ISSUE THREAD

**Issue by Self-Perfection** [dpurgin/harbour-callrecorder#51]
I pretty often do `systemctl --user restart lipstick` as it the fast way to reclaim RAM without full phone reboot (lots of services seems to leak memory). And after restart `harbour-callrecorderd` becomes not running, so I used manually start it each time. Quite annoying.

Journalctl from lipstick restart:
```
мар 27 18:03:11 Jolla harbour-callrecorderd[10510]: harbour-callrecorderd: Sun Mar 27 18:03:11 2016: [0x01c3c1a8]: qwaylandeventthread.cpp:81: void QWaylandEventThread::checkError() const: The Wayland connection broke. Did the Wayland compositor die?
мар 27 18:03:11 Jolla ohmd[410]: *** resource_request:  unregister   89
мар 27 18:03:12 Jolla kernel: mdp4_mixer_blend_setup: Error: no bg_pipe at mixer=0
мар 27 18:03:12 Jolla kernel: mipi_truly_otm9608a_lcd_off()
мар 27 18:03:12 Jolla kernel: mipi_truly_otm9608a_lcd_off()--
мар 27 18:03:12 Jolla systemd[861]: harbour-callrecorderd.service: main process exited, code=exited, status=1/FAILURE
мар 27 18:03:12 Jolla systemd[1]: Stopped Myriad Alien Dalvik.
мар 27 18:03:12 Jolla systemd[1]: Stopping Myriad Alien Dalvik Settings...
мар 27 18:03:12 Jolla systemd[861]: Unit harbour-callrecorderd.service entered failed state.
...
мар 27 18:03:13 Jolla harbour-callrecorderd[6332]: [D] QQmlDebuggingEnabler::QQmlDebuggingEnabler:1455 - QML debugging is enabled. Only use this in a safe environment.
мар 27 18:03:13 Jolla harbour-callrecorderd[6332]: [C] unknown:0 - Failed to create display (No such file or directory)
мар 27 18:03:13 Jolla systemd[861]: harbour-callrecorderd.service: main process exited, code=exited, status=1/FAILURE
мар 27 18:03:13 Jolla systemd[861]: Unit harbour-callrecorderd.service entered failed state.
```
There are a few more restarts in log and `harbour-callrecorder.serivce` settles in failed state before wayland becomes available.

I found a workaround: after I've moved symlink to service file from `~/.config/systemd/user/user-session.target.wants` to `~/.config/systemd/user/post-user-session.target.wants` `harbour-callrecorder.service` seems to start just fine on phone boot and on lipstick restart. So it seems tempting to write simple pull request to change target that wants the service.

However dependency of sound recording daemon on display availability (see logs above) seems odd and redundant. Is it possible to drop wayland dependency? I bet it is unused and getting rid of it would reduce daemon footprint.

**Comment by Alpherie** [comment]
Not sure if I am right, but in [here](https://github.com/dpurgin/harbour-callrecorder/blob/master/daemon/src/application.cpp#L177), when the daemon starts, GUI is created, and later, it only becomes visible when needed. That's why daemon is dependent on display availability.

**Comment by Self-Perfection** [comment]
Good point! I forgot about optional approval of storing recorded calls. But it seems to that creation of GUI on application start is a waste of resources. E.g. I just want to record every single call without being bothered by extra questions. IMHO GUI creation should be postponed to the moment just before it has to be shown for the first time.

**Comment by Alpherie** [comment]
Probably you are right, it is more logical to create the GUI on the actual moment when it is needed, and then destroy it after. Maybe even an option in settings can be added if the daemon should ask or save by default, so you would not be bothered by the question. (It would be not very difficult to implement)

On the other hand, solution now used can be a some kind of a fix for the problems we don't know.

**Comment by Self-Perfection** [comment]
> Maybe even an option in settings can be added if the daemon should ask or save by default, so you would not be bothered by the question.

[It already exists](https://github.com/dpurgin/harbour-callrecorder/blob/0.6.2/ui/qml/pages/settings/Storage.qml#L322) :) I have unchecked it long ago. That's why I forgot about this case which requires display access.

**Comment by Alpherie** [comment]
Yes, really exists. I searched for it in daemon settings, but it was in storage settings. It seems to me the GUI for the dialog is always created, not depending on this setting.

**Comment by dpurgin** [comment]
@Self-Perfection @Alpherie 
Guys, I've merged #52 and changed the .service file so that the daemon starts after lipstick. I cannot reproduce the bug anymore on Jolla C / Sailfish 2.0.2. Can you please build the app from rev. c4aa2948 and check if the problem is solved for you too? Thanks!

**Comment by Self-Perfection** [comment]
Hi dpurgin. I'm not sure it worth the effort. I have not tried 2.0.2 myself but changelog for 2.0.2 states, that there will be "Setting to enable automatic call recording". Is there a point in maintining you project of SfOS gets embedded call recorder?

**Comment by dpurgin** [comment]
It surely won't be, as soon as the embedded one becomes at least as feature rich as harbour-callrecorder. I've tried using the embedded one but as for now it has only two advantages - it's embedded and you can turn on and off the recording on the voicecall UI.

**Comment by d9h02f** [comment]
Major disadvantage is the embedded call recorder doesn't work on the Jolla phone (it records silence).

**Comment by Self-Perfection** [comment]
I see, it makes sense. Compiling from source will require some effort from me (e.g. I have not tried development for SailfishOS and therefore do not have SDK installed) but hopefully I will do it in a week or two. I was going to try making stuff for SfOS anyway.

**Comment by dpurgin** [comment]
Here is a zipped RPM I've build today from the latest source (GitHub doesn't allow RPM in attachment, sorry) 

[harbour-callrecorder-0.7.0-1.armv7hl.zip](https://github.com/dpurgin/harbour-callrecorder/files/437902/harbour-callrecorder-0.7.0-1.armv7hl.zip)

**Comment by Self-Perfection** [comment]
I have installed attached version today. Seems to work well in my use case. Successfully started after a couple of lipstick restarts.

I would like to note that previous release used about 7MB of RAM and this version mem footprint is reduced to about 2MB! I believe this is a result of 03d3ddddfd23f9c7d38e5507eb8eeac66a668db3 - lazy initialization of QML. I don't need confirmation dialog, so that is 5MB of RAM saved for me. Thank you!

Though workaround, that I proposed in first message, took more than a week to prove to be incomplete, I believe current shanges should resolve this issue completely and mark this issue resolved without lengthy testdrive.
